app/src/main/java/io/homeassistant/companion/android/background/LocationBroadcastReceiver.kt

```diff
@@ -208,6 +221,25 @@ class LocationBroadcastReceiver : BroadcastReceiver() {
         return geofencingRequestBuilder.build()
     }
 
+    private fun requestSingleAccurateLocation(context: Context) {
+        val request = createLocationRequest()
+        request.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+        request.numUpdates = 2 // Request 2 updates to make sure GPS gets a good lock
+        LocationServices.getFusedLocationProviderClient(context)
+            .requestLocationUpdates(
+                request,
+                object : LocationCallback() {
+                    override fun onLocationResult(locationResult: LocationResult?) {
```

Comment:
Will this be called twice?

Reply:
Yes, depending on how fast a GPS lock is acquired the first result is not always as accurate as we would want.  I figured letting it give us two updates should allow long enough for a decently accurate location to be sent.

Reply:
So this means we also send 2 location updates to Home Assistant, the first one being potentially inaccurate ?

Reply:
Correct, I thought about checking accuracy before sending this one as well.  However if we get the 2 updates and neither make the accuracy requirement then we really won't be sending an exit zone location at all and figured this would be better.

Reply:
I don't know if it's possible, but could we limit to to only send the last update ?